cognee/infrastructure/engine/models/Relationship.py

```diff
@@ -0,0 +1,70 @@
+from datetime import datetime, timezone
+from typing import Optional, Any, Dict
+from uuid import UUID, uuid4
+from pydantic import BaseModel, Field
+from typing_extensions import TypedDict
+import pickle
+
+# Define metadata type
+class RelationshipMetaData(TypedDict):
+    index_fields: list[str]
+
+
+class Relationship(BaseModel):
+    __tablename__ = "relationship"
```

Comment:
Edge-type embeddings are already implemented here:
https://github.com/topoteretes/cognee/pull/251

This collected the distinct edge types, creates a pedantic model for edge types: https://github.com/topoteretes/cognee/pull/251/files#diff-676c14525be801de96c5734e9d56bb784f3aaf40fc60d4f030a16e06f17317f9

and embed the relationship name into the edge_type collection.